routes/schema.js

```diff
@@ -0,0 +1,77 @@
+const { Joi } = require('celebrate')
+
+function depositAsset({ TOKEN_ADDRESS_WHITELIST }) {
+	let schema = Joi.string().required()
+	if (TOKEN_ADDRESS_WHITELIST && TOKEN_ADDRESS_WHITELIST.length > 0) {
+		schema = schema.valid(TOKEN_ADDRESS_WHITELIST)
+	}
+	return schema
+}
+
+function depositAmount({ MINIMAL_DEPOSIT }) {
+	let schema = Joi.number().required()
+	if (MINIMAL_DEPOSIT > 0) {
+		schema = schema.min(MINIMAL_DEPOSIT)
+	}
+	return schema
+}
+
+function creator({ CREATORS_WHITELIST }) {
+	let schema = Joi.array()
+		.items(Joi.string())
+		.required()
+		.length(2)
+	if (CREATORS_WHITELIST && CREATORS_WHITELIST.length > 0) {
+		schema = schema.has(CREATORS_WHITELIST)
+	}
+	return schema
+}
+
+function validators({ CREATORS_WHITELIST }) {
```

Comment:
also, other than `VALIDATORS_WHITELIST` (separate thing than CREATORS_WHITELIST), this should check if the validator count is always 2; not larger, not smaller, always `=== 2`

and when checking the whitelist, we should check VALIDATORS_WHITELIST OR `=== adapter.whoami()`

Reply:
yeah it already checks the validator count is always 2.